openssl/src/nid.rs

```diff
@@ -78,6 +78,22 @@ impl Nid {
         }
     }
 
+    /// Returns the `Nid` for the supplied `s` which can be the
+    /// numerical format, short name, or long name
+    #[corresponds(OBJ_txt2nid)]
+    pub fn txt_to_nid(s: &str) -> Result<Nid, ErrorStack> {
```

Comment:
Should this be a `Parse` implementation instead?

Reply:
Do you mean changing the function name to something like `from_str`, or what Parse implementation are you referencing?

Reply:
An important point about this function, it will not create that NID if it does not exist, so parse might be somewhat of a misnomer. It really is more like fetch nid if it exists

Reply:
Oh sorry, `FromStr`, not `Parse`: https://doc.rust-lang.org/stable/std/str/trait.FromStr.html

Reply:
I've updated to implement the `FromStr` trait for `Nid`